### DISCUSSION TITLE
Get Z value of mesh from GLTF

### DISCUSSION THREAD

**Discussion by DrAklinger** [bevyengine/bevy#13089] [Q&A]
Hello all,
currently, I am trying to get the z value from a loaded GLTF file which is my landscape. I need this to spawn different objects on top of the landscape (like stones, trees, ...). However, I can't figure out a good way on how to do this... 

Thanks in advance!

**Comment by GitGhillie** [comment]
You want the z value at specific points on the landscape right? Should this point be determined by the mouse position? If so, https://github.com/aevyrie/bevy_mod_picking might be an option

**Reply by DrAklinger** [reply]
Thank you for the reply - I already use this addon. But how do you think I could use that?

In my case, I need to determine the z value because some unit is 'driving' on top of the mesh -> can this be represented as 'mouse pointer'?

**Reply by GitGhillie** [reply]
No problem. `bevy_mod_picking` events like [Over](https://docs.rs/bevy_mod_picking/latest/bevy_mod_picking/events/struct.Over.html) contain [HitData](https://docs.rs/bevy_mod_picking/latest/bevy_mod_picking/backend/struct.HitData.html) which contains the position of where the mouse is hitting the mesh in world space. Does that help or are you looking something that is not necessarily driven by the mouse pointer?

**Reply by DrAklinger** [reply]
This one would not be driven by the mouse pointer. The scenario is that a few hundred units will walk on top of a mesh (aka landscape) created within blender. And in order to make them walk on top of the mesh, I would need to find out - given the x,y coordinates - what height (aka z) do I need to use.

**Reply by GitGhillie** [reply]
In that case you'd cast a ray from (x, y, z) in the -z direction and check where it hits the mesh. You'd pick the z coordinate of the ray origin such that it's guaranteed to start above the landscape.

To do raycasting against arbitrary geometry there's `bevy_mod_raycast`, which is also used by `bevy_mod_picking` as the default picking backend. Let me know if that helps!

**Reply by DrAklinger** [reply]
I guess I will give it a try, when I tackle "hilly" landscape again. But I wonder how good that scales with the amount of units..? Raytracing sounds expensive.

**Reply by GitGhillie** [reply]
If performance is a problem you could also generate a heightmap in advance and then sample points from there. I don't have experience in this direction though, and it wouldn't work with overhangs etc
